src/scf-release/src/acceptance-tests-brain/test-scripts/018_autoscaler_test.rb

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env ruby
+require_relative 'testutils'
+
+# APPS_DOMAIN   = <%= p('smoke_tests.apps_domain') %>
+# BROKER_URL    = <%= p('smoke_tests.autoscaler_service_broker_endpoint') %>
+# BROKER_USER   = <%= p('autoscaler.service_broker.username') %>
+# BROKER_PASS   = <%= p('autoscaler.service_broker.password') %>
+# SMOKE_SERVICE = <%= p('autoscaler.smoke.service_name') %>
+# SMOKE_PLAN    = <%= p('autoscaler.smoke.service_plan') %>
+
+APPS_DOMAIN   = ENV['APPS_DOMAIN']
+BROKER_USER   = ENV['BROKER_USER']
+BROKER_PASS   = ENV['BROKER_PASS']
+BROKER_URL    = ENV['BROKER_URL']
+SMOKE_SERVICE = ENV['SMOKE_SERVICE']
+SMOKE_PLAN    = ENV['SMOKE_PLAN']
+NAMESPACE     = ENV['KUBERNETES_NAMESPACE']
+
+STATEFULSETS = [ 'autoscaler-api',
+                 'autoscaler-eventgenerator',
+                 'autoscaler-metrics',
+                 'autoscaler-operator',
+                 'autoscaler-postgres',
+                 'autoscaler-scalingengine',
+                 'autoscaler-scheduler',
+                 'autoscaler-servicebroker' ]
+
+# ~ ~~ ~~~ ~~~~~ ~~~~~~~~ ~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~
+## Check if autoscaler pods are running
+
+have = 0
+STATEFULSETS.each do |name|
+  have += 1 if statefulset_ready(NAMESPACE, name)
+end
+
+# Skip the test if none of the autoscaler pods are running.
+puts "Autoscaler inactive." if have == 0
+exit_skipping_test if have == 0
+
+# Fail the test if some, but not all of the autoscaler pods are
+# running.
+fail "Autoscaler only partially active." if have < STATEFULSETS.length
+
+puts "Autoscaler active. Testing begins."
+STDOUT.flush
+
+# ~ ~~ ~~~ ~~~~~ ~~~~~~~~ ~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~
+## Standard cf cli access
+
+login
+setup_org_space
+
+# ~ ~~ ~~~ ~~~~~ ~~~~~~~~ ~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~
+##
+
+dora         = resource_path('dora')
+app_name     = random_suffix('dora')
+broker_name  = random_suffix('SCALER')
+service_name = random_suffix('SERVICE')
+url          = "http://#{app_name}.#{APPS_DOMAIN}"
+
+at_exit do
+  puts "Exiting. Cleanup."
+  STDOUT.flush
+  set errexit: false do
+    run "cf", "unbind-service", app_name, service_name
+    run "cf", "delete", "-f", app_name
+    run "cf", "delete-service", service_name, "-f"
+    run "cf", "delete-service-broker", broker_name, "-f"
+    end
+end
+
+run "cf", "push", app_name, "--no-start", "-p", dora
+run "cf", "create-service-broker", broker_name, BROKER_USER, BROKER_PASS, BROKER_URL
+run "cf", "service-access"
+run "cf", "enable-service-access", SMOKE_SERVICE, "-p", SMOKE_PLAN
+run "cf", "create-service", SMOKE_SERVICE, SMOKE_PLAN, service_name
```

Comment:
Is this a synchronous operation? For minibroker it is asynchronous.

Reply:
AFAIK sync. This code is straight from the original shell script.

Reply:
Nothing to do, nothing done